src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDao.java

```diff
@@ -86,13 +88,19 @@ public MandateSearchDao(Jdbi jdbi) {
         queryMap.put("gatewayAccountExternalId", params.gatewayAccountExternalId.getValue());
         params.getCaseInsensitivePartialSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey(), "%" + e.getValue() + "%"));
         params.getCaseInsensitivePartialSearchTermsOnPayer().forEach(e -> queryMap.put(e.getKey(), "%" + e.getValue() + "%"));
+        params.getExactSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey(), e.getValue()));
+        params.getGreaterThanSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey() + "_from", e.getValue()));
+        params.getLessThanSearchTermsOnMandate().forEach(e -> queryMap.put(e.getKey() + "_to", e.getValue()));
         return queryMap;
     }
 
     private String generateQuery(Params params) {
         StringBuilder sb = new StringBuilder();
         params.getCaseInsensitivePartialSearchTermsOnMandate().forEach(e -> sb.append(format(" AND m.%s ILIKE :%s", e.getKey(), e.getKey())));
         params.getCaseInsensitivePartialSearchTermsOnPayer().forEach(e -> sb.append(format(" AND p.%s ILIKE :%s", e.getKey(), e.getKey())));
+        params.getExactSearchTermsOnMandate().forEach(e -> sb.append(format(" AND m.%s = :%s", e.getKey(), e.getKey())));
```

Comment:
If we’ve already got a `StringBuilder`, we don’t really need to use `String.format(…)`, we can just do something like this:

```java
sb.append(" AND m.").append(e.getKey()).append(" = :").append(e.getKey());
```